sonar-batch/src/main/java/org/sonar/batch/issue/tracking/ServerIssueRepository.java

```diff
@@ -96,7 +96,7 @@ public Void apply(@Nullable ServerIssue issue) {
     profiler.stopDebug();
   }
 
-  public Iterable<ServerIssue> byComponent(BatchResource component) {
+  public Iterable<ServerIssue> byComponent(BatchComponent component) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png)The Cyclomatic Complexity of this method "byComponent" is 11 which is greater than 10 authorized.[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)